skimage/registration/tests/test_ilk.py

```diff
@@ -0,0 +1,108 @@
+import numpy as np
+from skimage._shared import testing
+from skimage.registration import optical_flow_ilk
+from skimage.transform import warp
+
+
+def _sin_flow_gen(image0, max_motion=4.5, npics=5):
+    """Generate a synthetic ground truth optical flow with a sinusoid as
+      first component.
+
+    Parameters:
+    ----
+    image0: ndarray
+        The base image to be warped.
+    max_motion: float
+        Maximum flow magnitude.
+    npics: int
+        Number of sinusoid pics.
+
+    Returns
+    -------
+    flow, image1 : ndarray
+        The synthetic ground truth optical flow with a sinusoid as
+        first component and the corresponding warped image.
+
+    """
+    grid = np.meshgrid(*[np.arange(n) for n in image0.shape], indexing='ij')
+    gt_flow = np.zeros_like(grid)
+    gt_flow[0, ...] = (np.sin(grid[0] / grid[0].max() * npics * np.pi)
+                       * max_motion)
+    image1 = warp(image0, grid - gt_flow, mode='nearest')
+    return gt_flow, image1
+
+
+def test_2d_motion():
+    # Generate synthetic data
+    rnd = np.random.RandomState(0)
+    image0 = rnd.normal(size=(256, 256))
+    gt_flow, image1 = _sin_flow_gen(image0)
+    # Estimate the flow
+    flow = optical_flow_ilk(image0, image1)
+    # Assert that the average absolute error is less then half a pixel
+    assert abs(flow - gt_flow) .mean() < 0.5
+
+
+def test_3d_motion():
+    # Generate synthetic data
+    rnd = np.random.RandomState(0)
+    image0 = rnd.normal(size=(128, 128, 128))
+    gt_flow, image1 = _sin_flow_gen(image0)
+    # Estimate the flow
+    flow = optical_flow_ilk(image0, image1)
+    # Assert that the average absolute error is less then half a pixel
+    assert abs(flow - gt_flow) .mean() < 0.5
+
+
+def test_no_motion_2d():
+    rnd = np.random.RandomState(0)
+    img = rnd.normal(size=(256, 256))
+
+    flow = optical_flow_ilk(img, img)
+
+    assert np.all(flow == 0)
+
+
+def test_no_motion_3d():
+    rnd = np.random.RandomState(0)
+    img = rnd.normal(size=(128, 128, 128))
+
+    flow = optical_flow_ilk(img, img)
+
+    assert np.all(flow == 0)
+
+
+def test_optical_flow_dtype():
+    # Generate synthetic data
+    rnd = np.random.RandomState(0)
+    image0 = rnd.normal(size=(256, 256))
+    gt_flow, image1 = _sin_flow_gen(image0)
+    # Estimate the flow at double precision
+    flow_f64 = optical_flow_ilk(image0, image1, dtype='float64')
+
+    assert flow_f64.dtype == 'float64'
```

Comment:
First time seeing that way to check dtypes. Seems a bit brittle and doesn't seem to work for cases such as `np.float64 == "float64"` which will return `False`. So maybe use `np.float64` here (and similar at other places) instead?

Reply:
You are right, `np.float64 == "float64"` returns `False`, but `np.dtype(np.float64) == 'float64'` is `True`. Here I compare data type object...
But I will check if this can be done differently ;-)

Reply:
`flow.dtype == np.float64` should work?

Reply:
Yes, absolutely :-)